sample_size/metrics.py

```diff
@@ -60,6 +67,15 @@ def _check_probability(probability: float) -> float:
         else:
             raise ValueError("Error: Please provide a float between 0 and 1 for probability.")
 
+    def generate_p_value(self, true_alt: bool, size: int, variants: int) -> Any:
+        if not true_alt:
+            return stats.uniform.rvs(0, 1, size=variants - 1)
+
+        else:
+            nc = np.sqrt(size / 2) * self.mde / self.variance
+            z_alt = stats.nct.rvs(nc=nc, df=2 * (size - 1), size=variants - 1)
+            return 2 * stats.norm.sf(np.abs(z_alt))
+
```

Comment:
I think the boolean metrics use a z-test, not a t-test, right?

```suggestion
    def generate_p_value(self, true_alt: bool, size: int, variants: int, n: int) -> Any:
        if not true_alt:
            return stats.uniform.rvs(0, 1, size=variants - 1)

        else:
            mu = self.mde / np.sqrt(self.variance / n)
            z_alt = stats.norm.rvs(loc=mu, size=variants - 1)
            return 2 * stats.norm.sf(np.abs(z_alt))

```

Reply:
This is something we can discuss. They are using z test as shown in line 77, but I still use `nct` when sampling their test statistics, because Kolmogorov–Smirnov tests' p-values are not uniformly distributed at all when I replaced

```
def direct_sample(δ, σ, n, size):
    nc = np.sqrt(n/2) * δ / σ
    return nct.rvs(nc=nc, df=2*(n-1), size=size)
```
by
```
def direct_sample(δ, σ, n, size):
    effect_size = δ / float(np.sqrt(2*σ**2/n))
    return stats.norm.rvs(loc=effect_size, size=size)
```
 in your script